### PR TITLE
Cleanups to run_tests.sh script

### PR DESCRIPTION
**Pull request by cbracken**
Bugfix:
* Use the `pub` from within the built Dart SDK (not whatever's on
  `$PATH`, if anything).

A few minor improvements:
* Allow running from below the src/ buildroot dir, as it's often
  convenient to work from within the flutter/engine git dir.
* Echo test name before running, for slightly better debuggability.
* Minor line-wrapping for readability.

### CODE REVIEW

File: testing/run_tests.sh [hunk 1]
```diff
@@ -2,35 +2,76 @@
...
+"$HOST_DIR/synchronization_unittests"
+
+echo "Running txt_unittests..."
+"$HOST_DIR/txt_unittests" --font-directory="$BUILDROOT_DIR/flutter/third_party/txt/third_party/fonts"
 
 # pubspec.yaml points to these files
-./flutter/tools/gn --unoptimized
-ninja -C out/host_debug_unopt flutter/sky/packages
+"$BUILDROOT_DIR/flutter/tools/gn" --unoptimized
+ninja -C $OUT_DIR/host_debug_unopt flutter/sky/packages
```
Comment (cbracken) [inline testing/run_tests.sh]: Is this intentionally hardcoded, or should it also be `$HOST_DIR`?

**Comment by cbracken** [comment]
FWIW, I plan to follow up with similar tweaks to `ci/test.sh` which also grabs whichever `dart` is on the path.

File: testing/run_tests.sh [hunk 2]
```diff
@@ -2,35 +2,76 @@
...
 
-pushd flutter/testing/dart
-pub get
+# Fetch Dart test dependencies.
+if [[ ! -x "$HOST_DIR/dart-sdk/bin/pub" ]]; then
```
Comment (mklim) [inline testing/run_tests.sh]: nit: Would it be better to maybe move this guard up to the top of the script so we fail on this before spending time on all the other tests?

Comment (cbracken) [reply testing/run_tests.sh]: Good point. Done. Simpler just to check for a Dart SDK dir as well.

File: testing/run_tests.sh (hunk 1 above)
Comment (cbracken) [reply testing/run_tests.sh]: Intentionally hardcoded due to pubspec.yaml path deps.

**Pull request merged**